sim.c

```diff
@@ -468,19 +468,19 @@ BEGIN_OPERATOR(delay)
   if (rate == 0)
     rate = 1;
   if (mod_num == 0)
-    mod_num = 10;
+    mod_num = 8;
   Glyph g = Tick_number % (rate * mod_num) == 0 ? '*' : '.';
   POKE(1, 0, g);
 END_OPERATOR
 
 BEGIN_OPERATOR(if)
   LOWERCASE_REQUIRES_BANG;
+  PORT(0, -1, IN);
   PORT(0, 1, IN);
-  PORT(0, 2, IN);
   PORT(1, 0, OUT);
-  Glyph g0 = PEEK(0, 1);
-  Glyph g1 = PEEK(0, 2);
-  POKE(1, 0, g0 == g1 ? '*' : '.');
+  Glyph g0 = PEEK(0, -1);
+  Glyph g1 = PEEK(0, 1);
+  POKE(1, 0, (g0 == g1 && g0 != '.' && g1 != '.') ? '*' : '.');
```

Comment:
Bad logic. if `g0 == g1` then there's no reason to test both `g0` and `g1` for inequality to `'.'`.

Reply:
`F` does not bang on equal `.` anymore.

Reply:
The other behavior of orca doesn't matter here. It's just simple boolean logic. If `g0` equals `g1`, and `g0` does not equal `'.'`, then `g1` also does not equal `'.'`.

Reply:
Oh I see, the extra `g1` check is unnecessary.

Reply:
ah yes you are right, i will make another commit for that!

Reply:
Just make sure to pull first, I have already made that change :)

Reply:
ah, perfect!